src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -106,13 +127,29 @@ public void toggleOrientation(){
         controlOrientationLog.append(controlOrientationIsFOD);
     }
 
+    public void enableRotationHold(int angle){
+        //Set the angle to automatically align the drive to using degrees -180 to 180
+        rotationHold = Units.degreesToRadians(angle);
+    }
+
+    public void disableRotationHold(){
+        rotationHold = null;
+    }
+
     public SwerveModuleState[] convertToModuleStates(double xTranslation, double yTranslation, double rotation) {
         //Takes axis input from joysticks and returns an array of swerve module states
 
         double x = yTranslation; //Intentional, x in swerve kinematics is y on the joystick
         double y = xTranslation;
         double r = rotation;
 
+        if (Math.abs(r) > 0){
```

Comment:
Do we need a deadband on this?

Reply:
No, since r already uses the deadband set in OI